### PR TITLE
🐛Source Recharge:  Fix owner slicing for metfaields stream

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
https://github.com/airbytehq/oncall/issues/1823 - changed the request_params() method to no longer apply stream_slice to the outbound query parameters. For one stream in particular, Metafields, we were taking a static list of ["customer", "store", "subscription"] which were injected into stream slices and would send out requests with ?owner_resource=customer query params. Because of how we reworked the request_params() method with slicing, these static values are included in the request to the Recharge API which causes the error.

## How
Fix slicing in metafields stream. 

## Pre-merge Checklist
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-recharge/source_recharge/api.py [hunk 1]
```diff
@@ -145,10 +144,23 @@ class Metafields(RechargeStream):
...
+        return params
+
+    def stream_slices(
+            self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
         owner_resources = ["customer", "store", "subscription"]
+
         for owner in owner_resources:
-            yield from super().read_records(stream_slice={"owner_resource": owner}, **kwargs)
+            yield {"owner_resource": owner}
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-recharge/source_recharge/api.py:163]: nit, but we could avoid the loop and create the slices and yield in fewer lines:
```
owner_resources = ["customer", "store", "subscription"]
yield from [{"owner_resource": owner} for owner in owner_resources]
```

File: airbyte-integrations/connectors/source-recharge/source_recharge/api.py [hunk 2]
```diff
@@ -35,7 +34,7 @@ def data_path(self):
         return self.name
 
     def path(
-        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-recharge/source_recharge/api.py:37]: These extra indentations don't look right. Are your pre-commit hooks not working right?

**Review by brianjlai — COMMENTED** [review]
A few nits and formatting suggestions. Otherwise looks good!

File: airbyte-integrations/connectors/source-recharge/source_recharge/api.py (hunk 2 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-recharge/source_recharge/api.py]: Started formatting manually. Should be better with extra indentations.

**Review by brianjlai — APPROVED** [review]
looks good to me!

**Pull request merged**